src/full_node/mempool_check_conditions.py

```
@@ -100,10 +102,41 @@ def mempool_assert_relative_time_exceeds(condition: ConditionVarPair, unspent: C
     return None
 
 
-def get_name_puzzle_conditions(block_program: SerializedProgram, safe_mode: bool):
-    # TODO: allow generator mod to take something (future)
-    # TODO: write more tests
-    block_program_args = SerializedProgram.from_bytes(b"\x80")
+def build_block_program_args(
+    clvm_deserializer: SerializedProgram, generator_refs: SerializedProgram
+) -> SerializedProgram:
+    """
+    The argument to the block program is a list. The first argument is the clvm
+    deserializer program, and the generators are from e.g FullBlock.generator_ref_list
+    (clvm_deserializer generator1 generator2 ...)
+    """
+
+    nil = NilSerializedProgram
+    if clvm_deserializer == nil or generator_refs == nil:
+        return nil
+
+    # TODO: Should the Generator block height ref list be in the form
+    # * A list of block heights: (int int int) or
+    # * A list of pairs of block heights, and generator byte string offsets: ((int . int) (int . int))
+    # `clvm_deserializer` is a clvm program, and generator_refs is a list as above
+    block_program_args = b"\xff" + bytes(clvm_deserializer) + bytes(generator_refs)
+
+    # TODO: Open the database, and read in the generators, inserting them in a new list,
+    # in the order requested in the reference list
+
+    # Note that if we fail to find a transaction block at the given height, we
+    # return a nil in that slot, so the program using these args is not given a list of
+    # a different length than expected.
```

I would say fail, that sounds like an invalid block if that's the case